lib/azurex/blob.ex

```diff
@@ -44,10 +45,34 @@ defmodule Azurex.Blob do
       {:error, %HTTPoison.Response{}}
 
   """
-  @spec put_blob(String.t(), binary, String.t(), optional_string, keyword) ::
+  @spec put_blob(
+          String.t(),
+          binary() | {:stream, Enumerable.t()},
+          optional_string,
+          optional_string,
+          keyword
+        ) ::
           :ok
           | {:error, HTTPoison.AsyncResponse.t() | HTTPoison.Error.t() | HTTPoison.Response.t()}
-  def put_blob(name, blob, content_type, container \\ nil, params \\ []) do
+  def put_blob(name, blob, content_type, container \\ nil, params \\ [])
+
+  def put_blob(name, {:stream, bitstream}, _content_type, container, params) do
```

Comment:
Is there any drawback to not setting the content type when (or after) streaming an asset?

Reply:
I don't believe so. Azure will set the content-type to `application/octet-stream` automatically if no type is specified when the block list is committed, which is our current behavior with MinIO. Though for the purposes of an open-source library, I should probably at least make it possible to provide a content-type if another user of this library wants that behavior